pkgs/development/libraries/hipfft/default.nix

```diff
@@ -0,0 +1,106 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rocmUpdateScript
+, cmake
+, rocm-cmake
+, hip
+, git
+, rocfft
+, gtest
+, boost
+, fftw
+, fftwFloat
+, openmp
+, buildTests ? false
+, buildBenchmarks ? false
+, buildSamples ? false
+}:
+
+# Can also use cuFFT
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hipfft";
+  version = "5.4.2";
+
+  outputs = [
+    "out"
+  ] ++ lib.optionals buildTests [
+    "test"
+  ] ++ lib.optionals buildBenchmarks [
+    "benchmark"
+  ] ++ lib.optionals buildSamples [
+    "sample"
+  ];
+
+  src = fetchFromGitHub {
+    owner = "ROCmSoftwarePlatform";
+    repo = "hipFFT";
+    rev = "rocm-${finalAttrs.version}";
+    hash = "sha256-yDtm9J0wqH6zo4HcgQbqhvwbzbOiJPQ48gJ2gC8PvjA=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    hip
+    git
+    cmake
+    rocm-cmake
+  ];
+
+  buildInputs = [
+    rocfft
+  ] ++ lib.optionals (buildTests || buildBenchmarks || buildSamples) [
+    gtest
+    boost
+    fftw
+    fftwFloat
+    openmp
+  ];
+
+  cmakeFlags = [
+    "-DCMAKE_C_COMPILER=hipcc"
+    "-DCMAKE_CXX_COMPILER=hipcc"
+    "-DCMAKE_MODULE_PATH=${hip}/lib/cmake/hip"
+    "-DHIP_ROOT_DIR=${hip}"
+    "-DHIP_PATH=${hip}"
+    # Manually define CMAKE_INSTALL_<DIR>
+    # See: https://github.com/NixOS/nixpkgs/pull/197838
```

Comment:
```suggestion
    # See: https://github.com/ROCm-Developer-Tools/hipamd/issues/55
```

Reply:
This is something that should be fixed upstream, in likely ``rocm-cmake``.
For not, there is no real reason to change this and said pull request explains the reasoning behind this.
See the current issue: https://github.com/RadeonOpenCompute/rocm-cmake/issues/121

Reply:
Then it should likely be this thread https://github.com/NixOS/nixpkgs/pull/197838#discussion_r1006537286 ?

Reply:
I think anyone who is curious enough to open that link can scroll down a bit honestly.
It's not worth a superfluous commit that will change sooner or later when this is fixed upstream.